pkg/apis/kubernetes/v1/cluster_types.go

```diff
@@ -31,6 +32,23 @@ type Cluster struct {
 	// Requires Service VMs.
 	EnableLBaaS *bool `json:"enable_lbaas,omitempty"`
 
+	// Identifier of an existing internal v4 prefix in which the cluster should be created.
+	// A new prefix is created if `ManageInternalIPv4Prefix` is true.
+	InternalIPv4Prefix *common.PartialResource `json:"internal_ipv4_prefix,omitempty"`
+	// Identifier of an existing external v4 prefix in which the cluster should be created.
+	// A new prefix is created if `ManageExternalIPv4Prefix` is true.
+	ExternalIPv4Prefix *common.PartialResource `json:"external_ipv4_prefix,omitempty"`
+	// Identifier of an existing external v6 prefix in which the cluster should be created.
+	// A new prefix is created if `ManageExternalIPv6Prefix` is true.
+	ExternalIPv6Prefix *common.PartialResource `json:"external_ipv6_prefix,omitempty"`
+
+	// If set to true an internal v4 prefix is automatically created for the cluster. Defaults to true if not set.
+	ManageInternalIPv4Prefix *bool `json:"manage_internal_ipv4_prefix,omitempty"`
+	// If set to true an external v4 prefix is automatically created for the cluster. Defaults to true if not set.
+	ManageExternalIPv4Prefix *bool `json:"manage_external_ipv4_prefix,omitempty"`
+	// If set to true an external v6 prefix is automatically created for the cluster. Defaults to true if not set.
+	ManageExternalIPv6Prefix *bool `json:"manage_external_ipv6_prefix,omitempty"`
```

Comment:
maybe we actually want to default them for Create calls in go-anxcloud, based on whether prefixes were configured?

Reply:
I think that could lead to confusion, as this would have too many hidden conditions.
If we don't limit the defaulting to Create operations, the following (pseudo!)code could easily "unmanage" prefixes by accident when updates are supported in the future.

```golang
cluster := kubernetesv1.Cluster{Identifier: "foo"}
api.Get(&cluster)
cluster.Name = "bar"
api.Update(&cluster)
```

Reply:
I think a different default logic on create would cause more confusion. Maybe we should instead return an error if prefix is set and managed on create?

Reply:
Returning an error sounds good!